src/libasr/codegen/asr_to_wasm.cpp

```diff
@@ -2084,8 +2084,9 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
     }
 
     void visit_StringConstant(const ASR::StringConstant_t &x) {
-        emit_string(x.m_s);
-        m_wa.emit_i32_const(m_string_to_iov_loc_map[x.m_s]);
+        std::string s = unescape_string(m_al, x.m_s);
```

Comment:
The string should have been unescaped in ASR already. See https://github.com/lcompilers/lpython/issues/1759. 

So the proper fix for this is to fix the frontend to keep the string unescaped.